### PR TITLE
Make onSnapshot work with rxjs observers

### PR DESCRIPTION
**Pull request by Feiyang1**
It fixes an issue where passing rxjs `Observer` to `onSnapshot` causes error `Error: Uncaught TypeError: this._next is not a function`.
For example,
```
const data$ = rxjs.Observable.create(function subscribe(observer) {
  firebase.firestore().collection('test').onSnapshot(observer);
});

data$.subscribe({
  next: data => handleData(data)
});
```

A similar issue is being fixed for FCM - https://github.com/firebase/firebase-js-sdk/pull/3221

### CODE REVIEW

File: .changeset/thin-ligers-fold.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"@firebase/firestore": patch
+"firebase": patch
+---
+
+Fixed an issue where `onSnapshot` doesn't work with rxjs [Observer](https://rxjs-dev.firebaseapp.com/guide/observer).
```
Comment (schmidt-sebastian) [inline .changeset/thin-ligers-fold.md:6]: FWIW, the change that broke this was merged on Friday and never released.

Comment (Feiyang1) [reply .changeset/thin-ligers-fold.md:6]: which one is it?

Comment (schmidt-sebastian) [reply .changeset/thin-ligers-fold.md:6]: https://github.com/firebase/firebase-js-sdk/pull/3294/files#diff-9cfbf72c75b0e1f9df5069543a6bced1R1172

Comment (Feiyang1) [reply .changeset/thin-ligers-fold.md]: Cool. Changed to an empty changeset.

**Pull request merged**